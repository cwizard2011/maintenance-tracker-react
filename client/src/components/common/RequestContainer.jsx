import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const RequestContainer = (props) => {
  const {
    title, label, name, status, requestId
  } = props;
  return (
    <div className="curved-border">
      <h3>{title}</h3>
      <p>
      Status:
        {' '}
        <span className={label}>
          <ion-icon name="icon" />
          {status}
        </span>
        <span className="create">
          <Link to={`/request/${requestId}`} className="view-btn">
            <ion-icon name={name} />
            Details
          </Link>

        </span>

      </p>
    </div>
  );
};

RequestContainer.propTypes = {
  title: PropTypes.string,
  label: PropTypes.string,
  name: PropTypes.string,
  status: PropTypes.string,
  requestId: PropTypes.string
};

export default RequestContainer;
