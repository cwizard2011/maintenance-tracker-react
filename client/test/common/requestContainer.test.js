import React from 'react';
import { shallow } from 'enzyme';
import RequestContainer from '../../src/components/common/RequestContainer';

describe('Guest Navigation', () => {
  it('render guest navigation correctly', () => {
    const wrapper = shallow(<RequestContainer />);
    expect(wrapper).toMatchSnapshot();
  });
});
