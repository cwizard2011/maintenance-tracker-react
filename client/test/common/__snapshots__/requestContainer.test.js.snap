// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guest Navigation render guest navigation correctly 1`] = `
<div
  className="curved-border"
>
  <h3 />
  <p>
    Status:
     
    <span>
      <ion-icon
        name="icon"
      />
    </span>
    <span
      className="create"
    >
      <Link
        className="view-btn"
        replace={false}
        to="/request/undefined"
      >
        <ion-icon />
        Details
      </Link>
    </span>
  </p>
</div>
`;
